x.py

```diff
@@ -104,6 +104,10 @@ def get_linux_env():
         ('JAVA_HOME', java_home),
         ('RUST_TEST_THREADS', '1'),
     ]
+    if java_home is None:
```

Comment:
I think we should check also for other operating systems.

Reply:
Done, I haven't been able to test on them but I think it should now raise an error for Windows and Mac as well